pkg/util/subnet/subnetsupdater.go

```diff
@@ -0,0 +1,69 @@
+package subnet
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+import (
+	"strings"
+
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+type Updater interface {
+	AddEndpointsToSubnets(endpoints []string, subnets []*mgmtnetwork.Subnet) (updatedSubnets []*mgmtnetwork.Subnet)
+}
+
+type UpdaterManager struct {
+}
+
+func (um *UpdaterManager) AddEndpointsToSubnets(endpoints []string, subnets []*mgmtnetwork.Subnet) (updatedSubnets []*mgmtnetwork.Subnet) {
+	for _, subnet := range subnets {
+		subnetChanged := addEndpointsToSubnet(endpoints, subnet)
+		if subnetChanged {
+			updatedSubnets = append(updatedSubnets, subnet)
+		}
+	}
+
+	return updatedSubnets
+}
+
+func addEndpointsToSubnet(endpoints []string, subnet *mgmtnetwork.Subnet) (subnetChanged bool) {
+	for _, endpoint := range endpoints {
+		endpointFound, serviceEndpointPtr := subnetContainsEndpoint(subnet, endpoint)
+		if !endpointFound ||
+			serviceEndpointPtr != nil && serviceEndpointPtr.ProvisioningState != mgmtnetwork.Succeeded {
```

Comment:
you should put some parenthesis on line 34. it may be the same, but we want to keep the exact same statements and not chain them like this

Reply:
Parenthesis are not needed  (Unit Tests are covering this and also this is just an "or" following an "and") but I have added an explanatory variable for readability.

Reply:
this is not about readability this is about the logic. if you don't put the parenthesis it means that it is evaluated like this 
`(!endpointFound || serviceEndpointPtr != nil ) && serviceEndpointPtr.ProvisioningState != mgmtnetwork.Succeeded `
So yes the result is the same , but the logic isn't.